### ISSUE TITLE
Some tests failing depending on timezone

### ISSUE THREAD

**Issue by sxhinzvc** [spring-projects/spring-data-mongodb#4446]
When running the build, the following tests are failing due to date mismatch, based on the timezone.

`MappingMongoConverterUnitTests#projectShouldReadSimpleInterfaceProjection`
`MappingMongoConverterUnitTests#projectShouldReadSimpleDtoProjection`
`AbstractEncryptionTestBase#encryptAndDecryptJavaTime`

The tests are asserting against LocalDate objects in my timezone, which is UTC-5.

For `MappingMongoConverterUnitTests#projectShouldReadSimpleInterfaceProjection`:
```
org.opentest4j.AssertionFailedError: 
expected: 1999-12-01 (java.time.LocalDate)
 but was: 1999-11-30 (java.time.LocalDate)
```

**Comment by christophstrobl** [comment]
Good catch - the testsetup is using `UTC` to define the source `date`. This is different to what the default `Jsr310Converters.LocalDateToDateConverter` does when writing a `LocalDateInstance` leading to the offset.

There's 2 options.

Use the `LocalDateToDateConverter` to create the source data.

```java
org.bson.Document source = new org.bson.Document("birthDate",
	LocalDateToDateConverter.INSTANCE.convert(LocalDate.of(1999, 12, 1)))
	.append("foo", "Walter");
```

or stick to the `UTC` date conversion and set up the converter differently to use native `java.time` codecs. 

```java
configureConverterWithNativeJavaTimeCodec();

org.bson.Document source = new org.bson.Document("birthDate",
	Date.from(LocalDate.of(1999, 12, 1).atStartOfDay().toInstant(ZoneOffset.UTC))).append("foo", "Walter");
```
